### PR TITLE
VOTE-1074: implement accessibility updates

### PR DESCRIPTION
**Pull request by mlloydbixal**
**_Delete all details that do not apply to this PR!_**

## Jira ticket (required)
[VOTE-823](https://bixal-projects.atlassian.net/browse/VOTE-823)
[VOTE-821](https://bixal-projects.atlassian.net/browse/VOTE-821)
[VOTE-820](https://bixal-projects.atlassian.net/browse/VOTE-820)

## Description (optional)
This PR adds missing aria-labels with their translations and improves the screen reader experience. 

### Pre-deploy steps
1. Insert step one

### Post-deploy steps
1. Insert step one

### Testing steps
1. Insert step one

### CODE REVIEW

File: web/themes/custom/vote_gov/templates/block/block-content--basic.html.twig [hunk 1]
```diff
@@ -2,23 +2,22 @@
 
 {% block content %}
 {% set classes = [
+    'bg-lightblue',
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/block/block-content--basic.html.twig:5]: We don't want to set this in all basic blocks. You can remove. We can add it to the classes field in the block

File: web/themes/custom/vote_gov/templates/block/block-content--email-signup--full.html.twig [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-            <label class="max-width-input text-grey" for="email-2" id="emailsub">
-                {{ content.field_email_field_label | field_value }}
-            </label>
-            <input id="email-2" name="email" type="email" class="usa-input max-width-input">
-            <button class="usa-button max-width-input" id="user-submit" type="submit">
-              {{ content.field_signup_button_label | field_value }}
-            </button>
-          </form>
-        </div>
+<section class="bg-white" aria-label="Email signup">
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/block/block-content--email-signup--full.html.twig:2]: The aria label needs to be translatable.

File: web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig [hunk 3]
```diff
@@ -60,7 +63,8 @@
           <p>
             <strong>{{ content.field_https_heading | field_value }}</strong>
             <br>
-            {% set lock_icon = '(<span class="icon-lock"><svg xmlns="http://www.w3.org/2000/svg" width="52" height="64" viewBox="0 0 52 64" class="usa-banner__lock-image">
+            {% set lock_icon = '(<span class="icon-lock"><svg xmlns="http://www.w3.org/2000/svg" width="52" height="64" viewBox="0 0 52 64" class="usa-banner__lock-image" role="graphics-symbol" aria-label="lock icon">
+                    <title>lock icon</title>
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig:67]: We need the aria label and the title to be translatable.

File: web/themes/custom/vote_gov/templates/layout/html.html.twig [hunk 4]
```diff
@@ -41,7 +42,8 @@
     <title>{{ head_title|safe_join(' | ') }}</title>
     <css-placeholder token="{{ placeholder_token }}">
       <js-placeholder token="{{ placeholder_token }}">
-        </head>
+  </head>
+</head>
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/layout/html.html.twig:46]: We only need one closing tag

File: web/themes/custom/vote_gov/templates/partial/footer-contact.html.twig [hunk 5]
```diff
@@ -1,11 +1,11 @@
-<div class="bg-lightblue footer-contact" data-test="footer-contact">
-
+<div class="bg-lightblue footer-contact" data-test="footer-contact" aria-label="Contact">
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/footer-contact.html.twig:1]: Aria label needs to be translatable.

File: web/themes/custom/vote_gov/templates/partial/footer-contact.html.twig [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
   <div class="grid-container eac-vote-partnership">
     <div class="grid-row">
       <div class="tablet:grid-col-auto">
         <div class="vote-txt">{{ 'Vote.gov' | t }}</div>
-        <img class="vote-logo" loading="lazy" width="130" height="160" src="{{ theme_path }}/img/ballot_box-blue.svg" alt="{{ 'Web design standards' | t }}">
-      </div>
+        <img class="vote-logo" loading="lazy" width="130" height="160" src="{{ theme_path }}/img/ballot_box-blue.svg" alt="{{ 'Illustration of voter box with checked ballot being inserted' | t }}" aria-label="Vote.gov">
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/footer-contact.html.twig:6]: aria label needs to be translatable.

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: web/themes/custom/vote_gov/src/js/external-links.js [hunk 7]
```diff
@@ -28,7 +28,7 @@
     if (externalLinks.length) {
       for (let i = 0; i < externalLinks.length; i++) {
         if (!matchesWhitelistedDomain(externalLinks[i].href)) {
-          const svg = `<svg aria-hidden="true" viewBox="3 3 18 18" width="10" height="10"><path d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z" fill="currentColor"></path></svg>`;
+          const svg = `<svg aria-hidden="true" viewBox="3 3 18 18" width="10" height="10" role="graphics-symbol" aria-label="external icon"><title>external icon</title><path d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z" fill="currentColor"></path></svg>`;
```
[the remark this answers is not in the document]
Comment (rayestrada) [reply web/themes/custom/vote_gov/src/js/external-links.js:31]: Did you create a ticket for this? Can you post the ticket number so I can close this?

File: web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig [hunk 8]
```diff
@@ -60,10 +63,11 @@
...
                     <path fill="#000000" fill-rule="evenodd" d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z">
                     </path>
-                  </svg></span>)' %}
+                  </svg></span>)' | replace({ 'lock icon' : 'lock icon' | t }) %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig:70]: I'm not sure what you trying to do here. Would it work if you just do it like this?
`</svg></span>)' | t %}`

**Review by rayestrada — CHANGES_REQUESTED** [review]

**Comment by rayestrada** [comment]
@mlloydbixal please fix the php linting errors as well.
You can run `lando coder-sniffer` locally to get an output of errors that need fixing.

File: web/modules/custom/vote_utility/translations/es.po [hunk 9]
```diff
@@ -19,6 +19,9 @@ msgstr "Seleccione el idioma"
 msgid "Go back"
 msgstr "Volver"
 
+msgid "Illustration of voter box with checked ballot being inserted to indicate voting action"
+msgid "Ilustración del acto del voto donde una boleta marcada está siendo insertada en una urna."
```
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply web/modules/custom/vote_utility/translations/es.po:23]: Removed period

File: web/modules/custom/vote_utility/translations/ypk.po [hunk 10]
```diff
@@ -19,6 +19,9 @@ msgstr "Ulu nakmiighhu"
 msgid "Go back"
 msgstr "Uteghten"
 
+msgid "Illustration of voter box with checked ballot being inserted to indicate voting action"
+msgid "Teruwiiq nakmikellghem salngaganeng igaq apeghiiqaq kananghtekanga kelgutilghii nakmikilleghmeng. "
```
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply web/modules/custom/vote_utility/translations/ypk.po:23]: It was included in the translation we received but probably not necessary. I've removed period and space.

File: web/modules/custom/vote_utility/translations/nv.po
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply web/modules/custom/vote_utility/translations/nv.po:1]: Ok, I'll remove this from the PR. Is there somewhere we can make note of which languages don't need screen reader translations?

File: web/themes/custom/vote_gov/src/js/external-links.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply web/themes/custom/vote_gov/src/js/external-links.js:31]: VOTE-1157

File: web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig (hunk 8 above)
Comment (mlloydbixal) [reply web/themes/custom/vote_gov/templates/block/block-content--government-banner.html.twig:70]: It's replacing the string for the aria-label and title of "lock icon" with it's translation. We don't have those translations yet. I created a ticket for implementing translations for aria-labels and titles (VOTE-1157). I could remove this from the PR and add it to that ticket to implement later?

**Review by rayestrada — DISMISSED** [review]

**Review by rayestrada — APPROVED** [review]

**Pull request merged**